pkg/minishift/registration/redhat.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright (C) 2016 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registration
+
+import (
+	"fmt"
+	"github.com/docker/machine/libmachine/drivers"
+	"github.com/docker/machine/libmachine/provision"
+	"strings"
+)
+
+func init() {
+	Register("Redhat", &RegisteredRegistrator{
+		New: NewRedhatRegistrator,
+	})
+}
+
+func NewRedhatRegistrator(d drivers.Driver) Registrator {
+	return &RedhatRegistrator{
+		SSHCommander: provision.GenericSSHCommander{Driver: d},
+		Driver:       d,
+	}
+}
+
+type RedhatRegistrator struct {
+	SSHCommander
+	Driver drivers.Driver
+}
+
+func (registrator *RedhatRegistrator) CompatibleWithDistribution() bool {
+	osReleaseOut, err := registrator.SSHCommand("sudo cat /etc/os-release")
```

Comment:
We are taking os-release info here without passing it registrator because anyway it will be on ssh command which need to be executed so better here.